src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -345,23 +338,21 @@ public ITransformer Fit(IDataView input)
                 view = new WordTokenizingEstimator(h, xfCols).Fit(view).Transform(view);
             }
 
-            if (tparams.NeedsRemoveStopwordsTransform)
+            if (tparams.UsePredefinedStopWordRemover)
             {
                 Contracts.Assert(wordTokCols != null, "StopWords transform requires that word tokenization has been applied to the input text.");
-                var xfCols = new StopWordsCol[wordTokCols.Length];
+                var xfCols = new StopWordsRemovingTransform.ColumnInfo[wordTokCols.Length];
                 var dstCols = new string[wordTokCols.Length];
                 for (int i = 0; i < wordTokCols.Length; i++)
                 {
-                    var col = new StopWordsCol();
-                    col.Source = wordTokCols[i];
-                    col.Name = GenerateColumnName(view.Schema, wordTokCols[i], "StopWordsRemoverTransform");
-                    dstCols[i] = col.Name;
-                    tempCols.Add(col.Name);
-                    col.Language = tparams.StopwordsLanguage;
+                    var tempName = GenerateColumnName(view.Schema, wordTokCols[i], "StopWordsRemoverTransform");
+                    var col = new StopWordsRemovingTransform.ColumnInfo(wordTokCols[i], tempName, tparams.StopwordsLanguage);
+                    dstCols[i] = tempName;
+                    tempCols.Add(tempName);
 
                     xfCols[i] = col;
                 }
-                view = tparams.StopWordsRemover.CreateComponent(h, view, xfCols);
+                view = new StopWordsRemovingEstimator(h, xfCols).Fit(view).Transform(view);
```

Comment:
>StopWordsRemovingEstimator [](start = 27, length = 26)

looks like this was the last one, right? Can we now convert `TextFeaturizingTransformer` to become a chain?

Reply:
not in this PR of course


---
In reply to: [235548815](https://github.com/dotnet/machinelearning/pull/1646#discussion_r235548815) [](ancestors = 235548815)